tests/basic/storage_pool_test.go

```diff
@@ -9762,3 +9762,106 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
 		AfterEachTest(contexts)
 	})
 })
+
+var _ = Describe("{PoolExpandRebalanceShutdownNode}", func() {
+	/*
+		1. create one pool
+		2. run ios
+		3. expand pool to higher size
+		4. while it is in progress shutdown and restart the node
+		5. let rebalance continue.
+		6. check if pending operation continues
+	*/
+
+	var testrailID = 0
+	// Testrail Description : while pool expand Rebalance is in progress ShutdownNode and check operation resumes
+	var runID int
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("PoolExpandRebalanceShutdownNode",
+			"Get Px State after pool delete",
+			nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+	stepLog := "Get Px State after pool delete"
+	It(stepLog, func() {
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("rebalanceshutdown-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		// Get Pool with running IO on the cluster
+		poolUUID, err := GetPoolIDWithIOs(contexts)
+		log.FailOnError(err, "Failed to get pool running with IO")
+		log.InfoD("Pool UUID on which IO is running [%s]", poolUUID)
+
+		// Get Node Details of the Pool with IO
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", poolUUID)
+		log.InfoD("Pool with UUID [%v] present in Node [%v]", poolUUID, nodeDetail.Name)
+
+		if IsLocalCluster(*nodeDetail) != true {
+			log.FailOnError(fmt.Errorf("This test will only support onprem vms"), "is this onprem?")
+		}
+		// Get Total Pools present on the Node present
+		poolDetails, err := GetPoolsDetailsOnNode(*nodeDetail)
+		log.FailOnError(err, "Failed to get Pool Details from Node [%v]", nodeDetail.Name)
+		log.InfoD("List of Pools present in the node [%v]", poolDetails)
+		poolToBeResized, err := GetStoragePoolByUUID(poolDetails[0].Uuid)
+		drvSize, err := getPoolDiskSize(poolToBeResized)
+		log.FailOnError(err, "error getting drive size for pool [%s]", poolToBeResized)
+		//add more disk since it should take longer for the resize to take longer
+		driveSize := drvSize * 3
+		expectedSize := (poolToBeResized.TotalSize / units.GiB) + driveSize
+		err = Inst().V.ExpandPoolUsingPxctlCmd(*nodeDetail, poolToBeResized.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, true)
```

Comment:
any particular reason you are using pxctl command?

Reply:
this is used to check if the volume is clean and then use it for expand

Reply:
how does this make sure volume is clean?

Reply:
true flag in the func

Reply:
ExpandPool function also has it, can you use that?

Reply:
Done